core/src/main/resources/reference.conf

```diff
@@ -12,37 +12,37 @@
 #   my-cassandra-journal {
 #     <settings...>
 #   }
+
+datastax-java-driver {
+  profiles {
+    cassandra-journal {
+      basic.request {
+        consistency = QUORUM
+        # the journal does not use any counters or collections
+        default-idempotence = true
+      }
+    }
+  }
+}
+
 cassandra-journal {
 
   # FQCN of the cassandra journal plugin
   class = "akka.persistence.cassandra.journal.CassandraJournal"
 
-  # List of contact points in the Cassandra cluster.
-  # Host:Port pairs are also supported. In that case the port parameter will be ignored.
-  # The value can be either a proper list, e.g. ["127.0.0.1", "127.0.0.2"],
-  # or a comma-separated list within a single string, e.g. "127.0.0.1,127.0.0.2".
-  contact-points = ["127.0.0.1"]
-
-  # Port of contact points in the Cassandra cluster.
-  # Will be ignored if the contact point list is defined by host:port pairs.
-  port = 9042
-
   # The implementation of akka.cassandra.session.SessionProvider
-  # is used for creating the Cassandra Session. By default the 
-  # the ConfigSessionProvider is building the Cluster from configuration properties
-  # but it is possible to replace the implementation of the SessionProvider
-  # to reuse another session or override the Cluster builder with other
-  # settings.
-  # For example, it is possible to lookup the contact points of the Cassandra cluster
-  # asynchronously instead of giving them in the configuration in a subclass of
-  # ConfigSessionProvider and overriding the lookupContactPoints method.
+  # is used for creating the CqlSession. By default it loads all the driver's default
+  # settings. The driver allows to override the way configuration is loaded
+  # see: https://docs.datastax.com/en/developer/java-driver/4.3/manual/core/configuration/#quick-overview
   # It may optionally have a constructor with an ActorSystem and Config parameter.
   # The config parameter is this config section of the plugin.
-  session-provider = akka.persistence.cassandra.ConfigSessionProvider
+  session-provider = "akka.cassandra.session.DefaultSessionProvider"
 
-  # The identifier that will be passed as parameter to the
-  # ConfigSessionProvider.lookupContactPoints method. 
-  cluster-id = ""
+  # profile to use, can set separate ones for reads and writes
+  # See https://docs.datastax.com/en/developer/java-driver/4.3/manual/core/configuration/
+  # for overriding any settings
+  read-profile = "cassandra-journal"
+  write-profile = "cassandra-journal"
```

Comment:
I guess this PR doesn't intend to solve entire https://github.com/akka/akka-persistence-cassandra/issues/81 ?
We still have cassandra-journal, cassandra-snapshot-store, cassandra-query-journal.

Good to do that as a separate change.

Reply:
Yes i'll leave until after, should be easy via an extension